File: Cpp/LeetCode125.ValidPalindrome.cpp
#include <string>

class Solution {
public:
    bool isPalindrome(std::string s) {
        int left =0;
        int right =s.length()- 1;
        while (left<right) {
            if (isalnum(s[left]) && isalnum(s[right])) {
                if (tolower(s[left]) != tolower(s[right])) {
                    return false;
                }
                left++;
                right--;
            }
            if (!isalnum(s[left]))
                left ++;
            
            if (!isalnum(s[right])) 
                right--;
        }
        return true;
    }
};

File: Python/LeetCode125.ValidPalindrome.py
import re


class Solution:
    def isPalindrome(self, s: str) -> bool:
        processed = re.sub(r'[\W_]', '', s.lower())
        left = 0
        right = len(processed) - 1
        while left < right:
            if processed[left] != processed[right]:
                return False
            left += 1
            right -= 1
        return True
